[Csc] CSC " + error.Message);

#### Analyzer package

[Microsoft.CodeAnalysis.FxCopAnalyzers](https://www.nuget.org/packages/Microsoft.CodeAnalysis.FxCopAnalyzers)

#### Package Version

v2.9.3 (Latest)

#### Repro steps

Hard to pinpoint that cause. During compilation an error is logged when running under teamcity...

[Csc] CSC " + error.Message);

Thats all the info I see, and it only affects a couple of our projects. Reverting to v2.9.2 fixes this issue.

#### Expected behavior

#### Actual behavior

I wish I could, this is all I see in the build log file from TC...

``
...`
[09:26:16]W:																 [Csc] Triggers\HolidayCalendarTrigger.cs(269, 30): warning CA1825: Avoid unnecessary zero-length array allocations.  Use Array.Empty<ICalendarEvent>() instead.
[09:26:16]W:																 [Csc] Triggers\HolidayCalendarTrigger.cs(261, 13): warning CA1031: Modify 'GetCurrentPublication' to catch a more specific exception type, or rethrow the exception.
[09:26:16]W:																 [Csc] Core\CalendarAdministration.cs(244, 27): warning CA2000: Call System.IDisposable.Dispose on object created by 'new DelegateResponseHandler<CalendarRequestAck>(
[09:26:16]E:																 [Csc] CSC " + error.Message);
[09:26:16]W:																 [Csc] Heartbeat\CalendarServerDiscovery.cs(18, 65): warning CA1507: Use nameof in place of string literal 'logger'
...
```

Changing back to 2.9.2 resolves this issue.

--- a/src/Microsoft.NetCore.Analyzers/Core/Runtime/DisposeObjectsBeforeLosingScope.cs
+++ b/src/Microsoft.NetCore.Analyzers/Core/Runtime/DisposeObjectsBeforeLosingScope.cs
@@ -1,5 +1,6 @@
 ﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
 
+using System;
 using System.Collections.Concurrent;
 using System.Collections.Immutable;
 using System.Linq;
@@ -223,7 +224,15 @@ namespace Microsoft.NetCore.Analyzers.Runtime
 
                     // CA2000: Call System.IDisposable.Dispose on object created by '{0}' before all references to it are out of scope.
                     var rule = GetRule(isNotDisposed);
+
+                    // Ensure that we do not include multiple lines for the object creation expression in the diagnostic message.
                     var argument = syntax.ToString();
+                    var indexOfNewLine = argument.IndexOf(Environment.NewLine, StringComparison.Ordinal);
+                    if (indexOfNewLine > 0)
+                    {
+                        argument = argument.Substring(0, indexOfNewLine);
+                    }
+
                     var diagnostic = syntax.CreateDiagnostic(rule, argument);
                     if (isNotDisposed)
                     {

--- a/src/Microsoft.NetCore.Analyzers/UnitTests/Runtime/DisposeObjectsBeforeLosingScopeTests.cs
+++ b/src/Microsoft.NetCore.Analyzers/UnitTests/Runtime/DisposeObjectsBeforeLosingScopeTests.cs
@@ -3,7 +3,6 @@
 using System;
 using System.Collections.Generic;
 using Analyzer.Utilities;
-using Microsoft.CodeAnalysis;
 using Microsoft.CodeAnalysis.CSharp;
 using Microsoft.CodeAnalysis.Diagnostics;
 using Microsoft.CodeAnalysis.Testing;
@@ -10327,6 +10326,34 @@ class C : IDisposable
 }", TestValidationMode.AllowCompileErrors,
             // Test0.cs(10,17): warning CA2000: Call System.IDisposable.Dispose on object created by 'new C()' before all references to it are out of scope.
             GetCSharpResultAt(10, 17, "new C()"));
+        }
+
+        [Fact, WorkItem(2529, "https://github.com/dotnet/roslyn-analyzers/issues/2529")]
+        public void MultilineDisposableCreation_SingleLine_Diagnostic()
+        {
+            VerifyCSharp(@"
+using System;
+
+class A : IDisposable
+{
+    public A(int a) { }
+    public void Dispose()
+    {
+    }
+}
+
+class Test
+{
+    void M1()
+    {
+        var a = new A(
+            M2());
+    }
+
+    int M2() => 0;
+}",
+            // Test0.cs(16,17): warning CA2000: Call System.IDisposable.Dispose on object created by 'new A(' before all references to it are out of scope.
+            GetCSharpResultAt(16, 17, "new A("));
         }
     }
 }

